library/ckagen.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include <mpi.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct kagen_gen_t;
```

Comment:
I would still suggest renaming this to just 'kagen' without '_gen'. Otherwise it is odd that 'kagen_create' returns a 'kagen_gen' pointer.

Reply:
This leads to name clashes, because we already have a namespace named `kagen`

Reply:
Oh, right. Should we use `kagen_t` then?

Reply:
I see, i misunderstood your comment and now the struct is named `kagen_gen_t`, but we use the alias `kagen_gen` in the public facing API. I just learned that C allows `typedef`ing a struct with the same name, so I will change it to
```c
typedef struct kagen_t {
} kagen_t

typedef struct kagen_result_t {
} kagen_result_t
```